server/s3_post_multipartobject_action.cc

```diff
@@ -524,6 +524,8 @@ void S3PostMultipartObjectAction::save_upload_metadata() {
   object_multipart_metadata->set_content_type(request->get_content_type());
   object_multipart_metadata->set_part_index_oid(
       part_metadata->get_part_index_oid());
+  object_multipart_metadata->regenerate_version_id();
```

Comment:
This action class is about 'initiate multipart'. In this request, S3 server never creates object, but it is still not live to external world. When Post complete request is sent to S3 server, the object created here will be committed or will go live to external world. It is only during post complete request, the object is actually live, and the version is created during post complete action.

So, this regenerate_version_id() is not required here, as this is not final post complete action that makes the object live.

Reply:
Ok, this regenerate_version_id() here will fix assert() issue that was observed during ST run.
Post complete action also calls regenerate_version_id() when object is actually becoming live.